src/rust/ensogl/lib/components/src/selector/number.rs

```diff
@@ -0,0 +1,108 @@
+        allow_click_selection(bool),
+        set_caption(Option<String>),
+        set_left_corner_round(bool),
+        set_right_corner_round(bool),
+    }
+    Output {
+        value(f32),
+        bounds(Bounds),
+    }
+}
+
+impl component::Frp<Model> for Frp {
+    fn init(&self, app:&Application, model:&Model, style:&StyleWatchFrp){
+        let frp     = &self;
+        let network = &frp.network;
+        let scene   = app.display.scene();
+        let mouse   = &scene.mouse.frp;
+
+        let base_frp = BaseFrp::new(model,style,network,frp.resize.clone().into(),mouse);
+
+        let track_shape_system = scene.shapes.shape_system(PhantomData::<track::Shape>);
+        track_shape_system.shape_system.set_pointer_events(false);
+
+        let background_click = relative_shape_click_position(model, network, &model.background.events, mouse);
+        let track_click      = relative_shape_click_position(model, network, &model.track.events, mouse);
+
+        frp::extend! { network
+
+                // Rebind
+                frp.source.bounds <+ frp.set_bounds;
+                frp.source.value  <+ frp.set_value;
+
+                // Simple Inputs
+                eval frp.set_caption((caption) model.set_caption_left(caption.clone()));
+                eval frp.set_left_corner_round ((value) model.left_corner_round(*value));
+                eval frp.set_right_corner_round((value) model.right_corner_round(*value));
+
+                // Intern State Vales
+                norm_value                 <- all2(&frp.value,&frp.bounds).map(normalise_value);
+                _has_overflow_bounds       <- frp.use_overflow_bounds.map(|value| value.is_none());
+                normalised_overflow_bounds <- all(&frp.use_overflow_bounds,&frp.bounds).map(|(overflow_bounds,bounds)|
+                    overflow_bounds.map(|(lower,upper)| (normalise_value(&(lower,*bounds)),normalise_value(&(upper,*bounds))) )
+                );
+                has_underflow <- norm_value.map(|value| *value < 0.0);
+                has_overflow  <- norm_value.map(|value| *value > 1.0);
+
+                // Value Updates
+                eval norm_value((value) model.set_background_value(*value));
+                eval has_underflow ((underflow) model.show_left_overflow(*underflow));
+                eval has_overflow ((overflow) model.show_right_overflow(*overflow));
+                eval frp.value((value) model.set_center_label_content(*value));
+
+                // Mouse IO
+                click <- any(&track_click,&background_click).gate(&frp.allow_click_selection);
+                click_value_update <- click.map2(&base_frp.track_max_width,position_to_normalised_value);
+
+                is_dragging <- any(
+                    base_frp.is_dragging_track,
+                    base_frp.is_dragging_background,
+                    base_frp.is_dragging_left_overflow,
+                    base_frp.is_dragging_right_overflow
+                );
```

Comment:
So far we use another formatting:
```suggestion
                is_dragging <- any
                    ( base_frp.is_dragging_track
                    , base_frp.is_dragging_background
                    , base_frp.is_dragging_left_overflow
                    , base_frp.is_dragging_right_overflow
                    );
```